miraclelinux/meta-debian
# libx11: Update version from 1.6.7-1+deb10u2 to 1.6.7-1+deb10u3

# Purpose of pull request
This version fixes following CVEs.
- CVE-2023-3138

DLA: DLA 3472-1

# Test
1. Build package
2. Run image and test the package on the qemuarm64 environment

## How to test the package
1. Build qemuarm64 image
2. Run qemuarm64 image
3. Check Xorg application uses libX11 library and works

### Build qemuarm64 image
Add the following to local.conf and build qemuarm64 image.
```
IMAGE_INSTALL_append = " libx11"
IMAGE_INSTALL_append = " xeyes"
EXTRA_IMAGE_FEATURES_append = " tools-sdk"
```
```
$ bitbake core-image-weston
```

### Run qemuarm64 image
```
$ runqemu qemuarm64
```

### Check Xorg application uses libX11 library and works
```
# ldd /usr/bin/weston-terminal | grep -i x11
# ldd /usr/bin/xeyes
# xeyes
```

## Test result
### Build package
Build succeeded.
```
$ bitbake libx11
Loading cache: 100% |############################################################################################################################################################################################################################################| Time: 0:00:00
Loaded 2384 entries from dependency cache.
Checking Debian Release ...
Sources.xz is not changed.
Force regenerate source code information.
Parsing Debian Sources.xz ...
Parsing recipes: 100% |##########################################################################################################################################################################################################################################| Time: 0:00:00
Parsing of 1362 .bb files complete (1361 cached, 1 parsed). 2384 targets, 71 skipped, 6 masked, 0 errors.
NOTE: Resolving any missing task queue dependencies

Build Configuration:
BB_VERSION           = "1.42.0"
BUILD_SYS            = "x86_64-linux"
NATIVELSBSTRING      = "universal"
TARGET_SYS           = "aarch64-emlinux-linux"
MACHINE              = "qemuarm64"
DISTRO               = "emlinux"
DISTRO_VERSION       = "2.7"
TUNE_FEATURES        = "aarch64 armv8a crc"
TARGET_FPU           = ""
meta                 
meta-yocto-bsp       = "HEAD:d4b57c68b22027c2bedff335dee06af963e4f8a8"
meta-debian          = "update-libx11_1.6.7-1+deb10u3:d46cfe02706e17b13427c6dc5bc8843a37ae3863"
meta-debian-extended = "HEAD:406c4711f35074d93f794b1b1416e9ff3fa009f0"
meta-emlinux         = "HEAD:6796fb6293c38ccc9497c6ab80390a35fb96dc58"
meta-emlinux-private = "HEAD:8207525289999d2c41983eb8155127b62621cdb1"

Initialising tasks: 100% |#######################################################################################################################################################################################################################################| Time: 0:00:00
Sstate summary: Wanted 6 Found 0 Missed 6 Current 110 (0% match, 94% complete)
NOTE: Executing SetScene Tasks
NOTE: Executing RunQueue Tasks
NOTE: Tasks Summary: Attempted 725 tasks of which 709 didn't need to be rerun and all succeeded.
```

### Package test
The weston-terminal and the xeyes uses libX11 library and works fine.
![image](https://github.com/miraclelinux/meta-debian/assets/118246107/d92e8366-f86e-4ac0-9aa4-f500c76b0b4d)
